### PR TITLE
VxCentralScan - move configuration endpoints to `grout`

### PR DESCRIPTION
**Pull request by adghayes**
## Overview

Moves a handful of endpoints to `grout`, particularly those around configuration since they were logically bunched together. Moved calls down the component hierarchy where possible. Along they way found a couple bugs and inconsistencies in this part of the codebase that needs some TLC, will call out in the comments.

## Testing Plan
- new automated testing
- manual testing sanity check

## Checklist

- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] ~~I have added a screenshot and/or video to this PR to demo the change~~
- [ ] ~~I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates~~

### CODE REVIEW

File: apps/central-scan/frontend/src/app_root.tsx
Comment (adghayes) [inline apps/central-scan/frontend/src/app_root.tsx]: I simplify the logic here, but it still has an infinite retry sort of bug identified in m17a QA which led to #3633. I address this in the final commit after manual testing.

File: apps/central-scan/frontend/src/app_root.tsx
Comment (adghayes) [inline apps/central-scan/frontend/src/app_root.tsx:444]: I manually confirmed that wrapping this method in a promise still means that errors get caught in the error boundary.

File: apps/central-scan/frontend/src/app_root.tsx
Comment (adghayes) [inline apps/central-scan/frontend/src/app_root.tsx]: See final commit for changes.

File: apps/central-scan/frontend/src/components/toggle_test_mode_button.tsx
Comment (adghayes) [inline apps/central-scan/frontend/src/components/toggle_test_mode_button.tsx]: our `getCanUnconfigure` method on the `store` already contains all this logic, so these extra checks are redundant and may allow us to create behavior in tests with non-existent combinations of parameters

File: apps/central-scan/frontend/src/app_root.tsx
Comment (adghayes) [inline apps/central-scan/frontend/src/app_root.tsx:363]: The idea of this commit is that we need to manage the configuration mutation more closely (to get our auto-configure and then hold off until you re-insert USB approach correct) _and_ use the mutation's state as a source of truth for component state.

Comment (adghayes) [reply apps/central-scan/frontend/src/app_root.tsx:363]: see: https://votingworks.slack.com/archives/C05BNBX5C3G/p1687800639587809
see: https://votingworks.slack.com/archives/C05BNBX5C3G/p1687884921336539

Comment (adghayes) [reply apps/central-scan/frontend/src/app_root.tsx:363]: @jonahkagan I was looking at the VxScan solution - does that work because the mutation resets when the unconfigured screen is unmounted?

File: apps/central-scan/frontend/src/screens/admin_actions_screen.test.tsx
Comment (jonahkagan) [inline apps/central-scan/frontend/src/screens/admin_actions_screen.test.tsx:135]: is this meant to say "calls backend" not "calls progress"?

File: apps/central-scan/frontend/src/app_root.tsx
Comment (jonahkagan) [reply apps/central-scan/frontend/src/app_root.tsx:363]: I'm not totally sure what you're asking. Maybe let's discuss?

My quick take (without having dug into this code), is that having a `useEffect` with `configureMutation` in the dependency array makes me nervous.

I think we want the same behavior for scan and central-scan, right? So I would guess that using the same logic from scan should work...

**Review by jonahkagan — COMMENTED** [review]
Wow, this is awesome work! Thanks for converting all of these endpoints and for the clear PR tour.

I'd like to resolve the unconfiguring issue from the last commit before approving.

File: apps/central-scan/backend/src/central_scanner_app.ts
Comment (adghayes) [reply apps/central-scan/backend/src/central_scanner_app.ts:234]: The guards in the frontend, in most of these cases, are buttons being disabled. Which I think we should definitely keep. So I'll go the `assert` path.

File: apps/central-scan/frontend/src/app_root.tsx
Comment (adghayes) [reply apps/central-scan/frontend/src/app_root.tsx:363]: The only difference between the two is that in VxCentralScan we show a sort of "success screen" afterwards whereas in VxScan I think we just jump to the next component. We could maintain the success screen with callbacks but I think it exists to remind the user to eject the USB, but moving forward VxCentralScan also needs continuous backup... so I'm in favor of just removing it.

File: apps/central-scan/frontend/src/api.ts
Comment (adghayes) [reply apps/central-scan/frontend/src/api.ts:96]: Since all the ballot data is not going through `grout` currently, it actually does not need to invalidate anything else.

File: apps/central-scan/backend/src/central_scanner_app.ts
Comment (adghayes) [reply apps/central-scan/backend/src/central_scanner_app.ts:257]: yeah, the historical name here has been zero (that's what it is at the `importer` and `store`) level and I was trying to maintain some connection, but I agree that `zero` is generally confusing. will move away from it here.

File: apps/central-scan/frontend/src/screens/admin_actions_screen.tsx
Comment (adghayes) [reply apps/central-scan/frontend/src/screens/admin_actions_screen.tsx:116]: Originally I had "Loading Admin Actions..." show but I didn't like the whole UI jumpiness. In 99% of cases there will not be override data and the default will be accurate.

The `?? undefined` is because it might be `null` coming from the backend and I just like to convert at the boundaries for consistency.

File: apps/central-scan/backend/src/central_scanner_app.ts
Comment (adghayes) [reply apps/central-scan/backend/src/central_scanner_app.ts:234]: 38aedde4d

File: apps/central-scan/backend/src/central_scanner_app.ts
Comment (adghayes) [reply apps/central-scan/backend/src/central_scanner_app.ts:101]: 03e610037

File: apps/central-scan/frontend/src/components/set_mark_thresholds_modal.tsx
Comment (adghayes) [reply apps/central-scan/frontend/src/components/set_mark_thresholds_modal.tsx:104]: 152fc78f1

File: apps/central-scan/backend/src/central_scanner_app.ts
Comment (adghayes) [reply apps/central-scan/backend/src/central_scanner_app.ts]: a004a818a

File: apps/central-scan/frontend/src/screens/admin_actions_screen.tsx
Comment (jonahkagan) [reply apps/central-scan/frontend/src/screens/admin_actions_screen.tsx:116]: Ok, thanks for explaining!

File: apps/central-scan/frontend/src/app_root.tsx
Comment (jonahkagan) [reply apps/central-scan/frontend/src/app_root.tsx:363]: Sounds good to me, maybe worth talking through with @mattroe

File: apps/central-scan/backend/test/helpers/setup_app.ts
Comment (adghayes) [reply apps/central-scan/backend/test/helpers/setup_app.ts]: d3ebbadc1

Comment (adghayes) [reply apps/central-scan/backend/test/helpers/setup_app.ts]: i think `promisify` didn't work here because it expects an `(error, value) =>` callback whereas `server.close` is just an `(error) =>` callback.  Didn't dig much.

Applied the change to VxScan as well.

File: apps/central-scan/frontend/src/screens/admin_actions_screen.test.tsx
Comment (adghayes) [reply apps/central-scan/frontend/src/screens/admin_actions_screen.test.tsx]: 4d79f44f2

File: apps/central-scan/frontend/src/app_root.tsx
Comment (adghayes) [reply apps/central-scan/frontend/src/app_root.tsx:363]: Sounds good. This thread is the biggest piece of feedback, I'll wait till I'm actually at my desk tomorrow to check in with folks and resolve. 

@kofi-q if you get to VxCentralScan UI work before this is merged, you may want to branch off of this PR.

File: apps/central-scan/frontend/src/app_root.tsx
Comment (kshen0) [inline apps/central-scan/frontend/src/app_root.tsx:468]: Is this check redundant? Should it be impossible to have an `electionDefinition` without a successful configure?

Comment (adghayes) [reply apps/central-scan/frontend/src/app_root.tsx:468]: Not redundant, because you could have configured the machine last time you powered on or before we reset the mutation. Will be reworking this to be more similar to VxScan anyway, though, so this logic will likely be removed.

File: apps/central-scan/frontend/src/app_root.tsx
Comment (jonahkagan) [inline apps/central-scan/frontend/src/app_root.tsx]: ✂️ 

Love to see this

**Review by jonahkagan — APPROVED** [review]

**Pull request merged**
